### DISCUSSION TITLE
Go datasource fails to process dependency located on self-hosted gitlab on http

### DISCUSSION THREAD

**Discussion by fogrye** [renovatebot/renovate#28503] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, dev-master

### Please tell us more about your question or problem

We are using self-hosted gitlab under VPN with custom domain which makes it impossible to use https because any browser will mark the certificate as invalid. So any requests are done via http and it works correct with renovate except go projects. I believe that issue is in [this line](https://github.com/renovatebot/renovate/blob/1266db7458d9316d9af0ab07d74fb3c4669a31c4/lib/modules/datasource/go/base.ts#L101C5-L101C54). I wonder if it possible to fix this line because renovate knows gitlab's base url part and should consider which type of connection is used. Anyway thx for such awesome project.

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Fetching gitlab.companyname/my_group/my_other_repo via GONOPROXY match (repository=my_group/my_repo)
DEBUG: GET https://gitlab.companyname/my_group/my_other_repo?go-get=1 = (code=ECONNREFUSED, statusCode=-1 retryCount=2, duration=2) (repository=my_group/my_repo)
DEBUG: Datasource connection error (repository=my_group/my_repo)
       "datasource": "go",
       "packageName": "gitlab.companyname/my_group/my_other_repo",
       "url": undefined,
       "errCode": "ECONNREFUSED"
DEBUG: Failed to look up go package gitlab.companyname/my_group/my_other_repo (repository=my_group/my_repo, packageFile=go.mod, dependency=gitlab.companyname/my_group/my_other_repo)
```

</details>

**Comment by rarkins** [comment]
See if https://docs.renovatebot.com/configuration-options/#insecureregistry works for this case

**Reply by fogrye** [reply]
As docs says that setting is for docker registry but Golang package manager checks git repo, I will try that but I believe it won't help. Anyway thx for your response.

**Reply by rarkins** [reply]
Yes if that URL is passed to git instead our http handler then it likely won't have any effect. Does go itself automatically try https and then fall back to http, or does it require configuration to do so?

**Reply by fogrye** [reply]
I wish it was part of GO because it would be solved by GOINSECURE env, but that happens before. Datasource calls "https://myrepo?go-get=1" to receive text "go get http://myrepo" and than provide that text to GO. So this is not an issue by go mod but bug in datasource. It would work ok with gitlab.com/github.com but not in my case.

**Reply by rarkins** [reply]
Can you set up a goproxy for Renovate to use, so that it can hide the complexity from Renovate?

**Reply by rarkins** [reply]
BTW if you set GOINSECURE env then maybe Renovate could look for that and prepend http instead of https?

**Reply by fogrye** [reply]
> BTW if you set GOINSECURE env then maybe Renovate could look for that and prepend http instead of https?

It would be the best solution for this issue. I'm not sure that GOPROXY can have authorisation and exposure of private packages is definitely not what we want.

**Reply by rarkins** [reply, marked as answer]
If you'd like to submit a PR adding support for GOINSECURE in Renovate's go datasource lookups then please go ahead.

**Reply by fogrye** [reply]
> If you'd like to submit a PR adding support for GOINSECURE in Renovate's go datasource lookups then please go ahead.

Could you pls convert it to issue, I will reference it in PR

**Reply by rarkins** [reply]
https://github.com/renovatebot/renovate/issues/28513
